base/commandLine.cpp

```diff
@@ -883,16 +901,28 @@ Common::String parseCommandLine(Common::StringMap &settings, int argc, const cha
 					usage("Non-existent extra path '%s'", option);
 				} else if (!path.isReadable()) {
 					usage("Non-readable extra path '%s'", option);
+				} else if (!path.isDirectory()
+				           && path.getParent().isDirectory()
+				           && path.getParent().isReadable()) {
+					// The path.getParent().isDirectory() check is redundant but included for clarifying
+					// the purpose of this case, which is to get the folder path part from a file path
+					settings["extrapath"] = path.getParent().getPath().toString(Common::Path::kNativeSeparator);
 				}
 			END_OPTION
 
 			DO_LONG_OPTION("iconspath")
-			Common::FSNode path(option);
-			if (!path.exists()) {
-				usage("Non-existent icons path '%s'", option);
-			} else if (!path.isReadable()) {
-				usage("Non-readable icons path '%s'", option);
-			}
+				Common::FSNode path(option);
+				if (!path.exists()) {
+					usage("Non-existent icons path '%s'", option);
+				} else if (!path.isReadable()) {
+					usage("Non-readable icons path '%s'", option);
+				} else if (!path.isDirectory()
+				           && path.getParent().isDirectory()
+				           && path.getParent().isReadable()) {
+					// The path.getParent().isDirectory() check is redundant but included for clarifying
+					// the purpose of this case, which is to get the folder path part from a file path
+					settings["iconspath"] = path.getParent().getPath().toString(Common::Path::kNativeSeparator);
```

Comment:
Since this code is repeated, could we wrap it into a reusable function?

Reply:
I couldn't come up with something pretty for a reusable function. 

I did a few iterations on implementing one, but the different cases per path option (isReadable() check or isWriteable check), the custom usage() messages (which should stay outside the function) and the updating of the settings[] value for the option, made the re-usable method complex and ultimately reduced readability of the code, rather than helped.

Any suggestions are of course welcome. 

For the record, my monstrosity looked like this:
`// Assumes that path.exists() check is true`
`bool ensureAccessibleDirectoryForPathOption(Common::StringMap &settings, const Common::String optionKeyStr, const Common::FSNode &path, bool ensureWriteable) {`
`	if (path.isDirectory()) {`
`		if ((ensureWriteable && path.isWritable())`
`		    || (!ensureWriteable && path.isReadable()))  {`
`			return true;`
`		}`
`	} else if (path.getParent().isDirectory()`
`	           && ((ensureWriteable && path.getParent().isWritable()) `
`	                || (!ensureWriteable && path.getParent().isReadable())) ) {`
`		// The path.getParent().isDirectory() check is redundant but included for clarifying`
`		// the purpose of this case, which is to get the folder path part from a file path`
`		settings[optionKeyStr] = path.getParent().getPath().toString(Common::Path::kNativeSeparator);`
`		return true;`
`	}`
`	return false;`
`}`

So that the parsing for eg. the screenshotpath would look like:
`			DO_LONG_OPTION("screenshotpath")`
`				Common::FSNode path(option);`
`				if (!path.exists()) {`
`					usage("Non-existent screenshot path '%s'", option);`
`				} else if (!ensureAccessibleDirectoryForPathOption(settings, "screenshotpath", path, true)) {`
`					usage("Non-writable screenshot path '%s'", option);`
`				}`
`			END_OPTION`

Reply:
> Since this code is repeated, could we wrap it into a reusable function?

I went ahead an committed the method above, with some comments to clarify its purpose